ChiModules/LinearBoltzmannSolver/lbs_03e_tgdsa.cc

```diff
@@ -1,188 +1,180 @@
-                                                    double *ref_phi_old,
-                                                    double *ref_phi_new)
+                                                    const double *ref_phi_old,
+                                                    const double *ref_phi_new)
 {
-  int gsi = groupset.groups[0].id;
-  int gss = groupset.groups.size();
+  const auto& sdm = *discretization;
+  const auto& phi_uk_man  = flux_moments_uk_man;
 
-  delta_phi_local.resize(local_node_count, 0.0);
+  const int    gsi = groupset.groups.front().id;
+  const size_t gss = groupset.groups.size();
+
+  delta_phi_local.assign(local_node_count, 0.0);
 
-  int index = -1;
   for (const auto& cell : grid->local_cells)
   {
-    int c = cell.local_id;
-
-    auto& transport_view = cell_transport_views[c];
+    const auto& cell_mapping = sdm.GetCellMapping(cell);
+    const size_t num_nodes = cell_mapping.NumNodes();
+    const auto& S = matid_to_xs_map[cell.material_id]->transfer_matrices[0];
 
-    chi_math::SparseMatrix& S = matid_to_xs_map[cell.material_id]->transfer_matrices[0];
-
-    for (int i=0; i < cell.vertex_ids.size(); i++)
+    for (size_t i=0; i < num_nodes; ++i)
     {
-      index++;
-      int m = 0;
-      size_t mapping = transport_view.MapDOF(i,m,0); //phi_new & old location gsi
+      const int64_t dphi_map = sdm.MapDOFLocal(cell, i);
+      const int64_t  phi_map = sdm.MapDOFLocal(cell, i,  phi_uk_man, 0, 0);
 
-      double* phi_old_mapped = &ref_phi_old[mapping];
-      double* phi_new_mapped = &ref_phi_new[mapping];
+            double& delta_phi_mapped = delta_phi_local[dphi_map];
+      const double* phi_old_mapped   = &ref_phi_old[phi_map];
+      const double* phi_new_mapped   = &ref_phi_new[phi_map];
 
-      for (int g=0; g<gss; g++)
+      for (size_t g=0; g<gss; ++g)
       {
         double R_g = 0.0;
-        int num_transfers = S.rowI_indices[gsi + g].size();
-        for (int j=0; j<num_transfers; j++)
-        {
-          int gp = S.rowI_indices[gsi + g][j];
-
-          if (gp < gsi + g + 1)
-            continue;
+        for (const auto& [row_g, gprime, sigma_sm] : S.Row(gsi+g))
+          if (gprime >= gsi + g)
+            R_g += sigma_sm * (phi_new_mapped[gprime] - phi_old_mapped[gprime]);
 
-          double delta_phi = phi_new_mapped[gp] - phi_old_mapped[gp];
-
-          R_g += S.rowI_values[gsi + g][j] * delta_phi;
-        }
-        delta_phi_local[index] += R_g;
+        delta_phi_mapped += R_g;
       }//for g
-
-    }//for dof
+    }//for node
   }//for cell
-
 }
 
 //###################################################################
 /**DAssembles a delta-phi vector on the first moment.*/
 void lbs::SteadySolver::DisAssembleTGDSADeltaPhiVector(LBSGroupset& groupset,
                                                        double *ref_phi_new)
 {
-  int gsi = groupset.groups[0].id;
-  int gss = groupset.groups.size();
+  const auto& sdm = *discretization;
```

Comment:
shouldn't this be sdm_ptr ??

Reply:
Nah. `discretization` is a smart pointer already. I will take on a killing spree soon and name things `grid_ptr` when its a smart pointer and `grid` when it's a reference. Same with spatial discretizations.